lib/infer/return.js

```diff
@@ -29,9 +31,14 @@ module.exports = function () {
     if (t.isFunction(fn) &&
       fn.returnType &&
       fn.returnType.typeAnnotation) {
-      comment.returns = [{
-        type: flowDoctrine(fn.returnType.typeAnnotation)
-      }];
+      var returnType = flowDoctrine(fn.returnType.typeAnnotation);
+      if (comment.returns) {
```

Comment:
Hm, good point. I'm not sure that that returns: [] ever happens in practice, but I'll add a guard here just to be sure.